goinstant/docker_test.go

```diff
@@ -0,0 +1,948 @@
+			element:  "Megatron",
+			result:   false,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.testInfo, func(t *testing.T) {
+			ans := sliceContains(tt.slice, tt.element)
+
+			if ans != tt.result {
+				t.Fatal("SliceContains should return" + fmt.Sprintf("%t", tt.result) + "but returned" + fmt.Sprintf("%t", ans))
+			}
+			t.Log(tt.testInfo + " passed!")
+		})
+	}
+}
+
+func Test_getPackagePaths(t *testing.T) {
+	type args struct {
+		inputArr []string
+		flags    []string
+	}
+	tests := []struct {
+		name             string
+		args             args
+		wantPackagePaths []string
+	}{
+		{
+			name: "Test 1 - '-c' flag",
+			args: args{
+				inputArr: []string{"-c=../docs", "-c=./docs"},
+				flags:    []string{"-c=", "--custom-package="},
+			},
+			wantPackagePaths: []string{"../docs", "./docs"},
+		},
+		{
+			name: "Test 2 - '--custom-package' flag",
+			args: args{
+				inputArr: []string{"--custom-package=../docs", "--custom-package=./docs"},
+				flags:    []string{"-c=", "--custom-package="},
+			},
+			wantPackagePaths: []string{"../docs", "./docs"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if gotPackagePaths := getPackagePaths(tt.args.inputArr, tt.args.flags); !assert.Equal(t, tt.wantPackagePaths, gotPackagePaths) {
+				t.Errorf("getPackagePaths() = %v, want %v", gotPackagePaths, tt.wantPackagePaths)
+			}
+		})
+	}
+}
+
+func Test_getEnvironmentVariables(t *testing.T) {
+	type args struct {
+		inputArr []string
+		flags    []string
+	}
+	tests := []struct {
+		name                     string
+		args                     args
+		wantEnvironmentVariables []string
+	}{
+		{
+			name: "Test case environment variables found",
+			args: args{
+				inputArr: []string{"-e=NODE_ENV=PROD", "-e=DOMAIN_NAME=instant.com"},
+				flags:    []string{"-e=", "--env-file="},
+			},
+			wantEnvironmentVariables: []string{"-e", "NODE_ENV=PROD", "-e", "DOMAIN_NAME=instant.com"},
+		},
+		{
+			name: "Test case environment variables file found",
+			args: args{
+				inputArr: []string{"--env-file=../test.env"},
+				flags:    []string{"-e=", "--env-file="},
+			},
+			wantEnvironmentVariables: []string{"--env-file", "../test.env"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if gotEnvironmentVariables := getEnvironmentVariables(tt.args.inputArr, tt.args.flags); !assert.Equal(t, tt.wantEnvironmentVariables, gotEnvironmentVariables) {
+				t.Errorf("getEnvironmentVariables() = %v, want %v", gotEnvironmentVariables, tt.wantEnvironmentVariables)
+			}
+		})
+	}
+}
+
+func Test_extractCommands(t *testing.T) {
+	type resultStruct struct {
+		environmentVariables []string
+		deployCommand        string
+		otherFlags           []string
+		targetLauncher       string
+		packages             []string
+		customPackagePaths   []string
+		instantVersion       string
+	}
+
+	testCases := []struct {
+		startupCommands []string
+		expectedResults resultStruct
+		testInfo        string
```

Comment:
Some of my tests use the default field 'name' as what you have as 'testInfo'. We should decide on a convention and stick to it.

Reply:
Same with default field 'args' for specifying arguments to the function being tested. Let's decide on a convention and stick to it.

Reply:
I see you changed from testInfo to name. Will update for consistency